Source/Internal/IGListAdapter+UICollectionView.m

```diff
@@ -57,6 +57,69 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
 
     return view;
 }
+    
+- (BOOL)collectionView:(UICollectionView *)collectionView canMoveItemAtIndexPath:(NSIndexPath *)indexPath {
+    id <IGListAdapterDataSource> dataSource = self.dataSource;
+    IGAssert(dataSource != nil, @"Found a nil dataSource when requesting canMoveItemAtIndexPath for interactive reordering");
+    
+    NSInteger sectionIndex = indexPath.section;
+    NSInteger itemIndex = indexPath.item;
+    
+    // for ease of implementation, first we check with the listAdapter
+    if ([dataSource respondsToSelector:@selector(listAdapter:canMoveObjectInSection:atIndex:)]) {
+        return [dataSource listAdapter:self canMoveObjectInSection:sectionIndex atIndex:itemIndex];
+    }
```

Comment:
I left a note above, but wondering if we can just skip this entirely? Maybe just rely on the section controller deciding if a move is allowed or not?

Reply:
Sure -- I was thinking about whole section reordering here. My motivation was that for section controllers to be reusable, they might be re-orderable or not depending on the context of the collection/adapter they appear in.

But, if the delegates are all to be required eventually, yes, it's worth dropping this to reduce complexity.